### PR TITLE
Setting up iD editor

### PR DESCRIPTION
**Pull request by Rub21**
cc @batpad

### CODE REVIEW

File: iD/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+FROM ubuntu:16.04
+ENV workdir /app
+RUN apt-get update -y
+RUN apt-get install -y curl
+RUN curl -sL https://deb.nodesource.com/setup_8.x | bash -
+RUN apt-get install -y nodejs
+RUN apt-get install -y wget python-pip software-properties-common python-software-properties git 
+RUN pip install awscli
+RUN git clone --depth=1 https://github.com/openstreetmap/iD.git $workdir
```
Comment (batpad) [inline iD/Dockerfile]: Can we pin this to a particular git sha so that we don't get unexpected changes?

**Comment by batpad** [comment]
@Rub21 so sorry for the late review here --

This looks great! I think the extra step of getting the oauth key, etc. and updating the deploy is fine for now, and the documentation of the process looks good.

We should also add  the Kubernetes template files to deploy this into the kubernetes cluster as well.

We would need something like a `deployment` and `service` like https://github.com/developmentseed/osm-seed/blob/develop/helm/osm-seed/templates/web-deployment.yaml and https://github.com/developmentseed/osm-seed/blob/develop/helm/osm-seed/templates/web-service.yaml .

We may need to think about how we configure different URLs and routing.

Let me know if you want to talk about this - I'm also okay with merging this as is, and doing the Kubernetes setup as a separate PR.

(have left one minor comment - it would be nice to pin the version of iD to an exact commit, so that iD doesn't unexpectedly change versions when we deploy / upgrade and cause problems.)

**Comment by batpad** [comment]
@Rub21 two things  blocking merge here:

 - [x] We should pin the version of `iD` that gets pulled to a specific commit.
 - [x] We should add this container to `dockerhub` and to the tests in CircleCI.

Happy to work through both of these with you.

**Comment by Rub21** [comment]
@batpad I think this is ready to merge!!

**Review by batpad — APPROVED** [review]

**Comment by majdal** [comment]
👏🏽👏🏽👏🏽👏🏽👏🏽

**Comment by majdal** [comment]
Hello hello - I'm trying out the iD editor, and there's something that's not 100% working. When I click on the save button, I get an about:blank popup, and nothing else happens. [Here's](https://www.dropbox.com/s/jl1e75su1f8nsli/Screen%20Shot%202018-09-08%20at%203.36.17%20PM.png?dl=0) a screenshot.

I'm not familiar enough with iD editor to know if this is an editor bug or an osm-seed bug. I'll try to investigate further, but I thought I should also ask here.

**Comment by Rub21** [comment]
Hey @majdal  have you changed  the OAUTH: https://github.com/developmentseed/osm-seed/blob/develop/.env.example#L38-L39, 

I think  you need to check these items 👇 

- Create a user in the DB.
- Create the [Oauth ids](https://user-images.githubusercontent.com/1152236/45333101-f2702800-b539-11e8-859b-b2313c0afa6f.png)
- Set the auth ids: https://github.com/developmentseed/osm-seed/blob/develop/.env.example#L38-L39
- Redeploy the app. 
  - (docker)`docker-compose down && docker-compose build && docker-compose up`  
  - (kubernetes)`helm upgrade -f osm-seed/value.yaml <name of the app> osm-seed/`

Also make sure, the IP or domain should be set : https://github.com/developmentseed/osm-seed/blob/develop/.env.example#L2

**Pull request merged**